### PR TITLE
Improve descriptions in Swarm-HDF-Spark wkws + exit if empty node sou…

### PR DESCRIPTION
**Pull request by mbenguig**
…rce name in Swarm-Spark and Swarm-HDFS-Spark wkws + add new enpoints to Spark

### CODE REVIEW

File: DockerSwarm/resources/catalog/Docker_Swarm.xml [hunk 1]
```diff
@@ -8,7 +8,7 @@
     <variable name="TARGETED_NB_PA_NODES" value="3" model="PA:NOT_EMPTY_STRING"/>
   </variables>
   <description>
-    <![CDATA[ Deploy a Docker Swarm. TARGETED_PA_NODE_SOURCE_NAME refers to the targeted node source form which TARGETED_NB_PA_NODES ProActive nodes will be reserved for the Docker Swarm. ]]>
+    <![CDATA[ Deploy a Docker Swarm. TARGETED_PA_NODE_SOURCE_NAME refers to the targeted node source from which TARGETED_NB_PA_NODES ProActive nodes will be reserved for the Docker Swarm. ]]>
```
Comment (nebilbenmabrouk) [inline DockerSwarm/resources/catalog/Docker_Swarm.xml:11]: Deploy a Docker Swarm network ... will be reserved for Docker Swarm

File: Spark/resources/catalog/Docker_Swarm_Spark_Service.xml [hunk 2]
```diff
@@ -1,21 +1,50 @@
...
+      <metadata>
+        <positionTop>
+          239
+        </positionTop>
+        <positionLeft>
+          451.5
+        </positionLeft>
+      </metadata>
+    </task>
+
```
Comment (nebilbenmabrouk) [inline Spark/resources/catalog/Docker_Swarm_Spark_Service.xml:47]: Why not simply a groovy pre-script in the first task? The error handling will be hidden, which is better

**Review by nebilbenmabrouk — APPROVED** [review]
Two minor remarks

File: DockerSwarm/resources/catalog/Docker_Swarm.xml (hunk 1 above)
Comment (mbenguig) [reply DockerSwarm/resources/catalog/Docker_Swarm.xml:11]: I prefer Create a Docker Swarm. There is no new created network from this workflow. Only from HDFS, Spark,...

File: Spark/resources/catalog/Docker_Swarm_Spark_Service.xml (hunk 2 above)
Comment (mbenguig) [reply Spark/resources/catalog/Docker_Swarm_Spark_Service.xml:47]: I prefer to isolate it. sorry.

**Pull request merged**